io-native/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -121,6 +143,11 @@ public static Object write(BObject channel, ArrayValue content) {
     public static Object close(BObject channel) {
         DelimitedRecordChannel recordChannel = (DelimitedRecordChannel) channel.getNativeData(TXT_RECORD_CHANNEL_NAME);
         try {
+            if(channel.getNativeData(BUFFERED_READER_ENTRY) != null) {
+                BufferedReader bufferedReader = (BufferedReader)
+                        channel.getNativeData(BUFFERED_READER_ENTRY);
+                bufferedReader.close();
+            }
```

Comment:
Can be moved to same function

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/31/commits/98acb67cdbfd15b787de2bc02c14a106f0615bfe